### ISSUE TITLE
To do GeneralizedRCNNTrasnform.py similar to Detectron2 

### ISSUE THREAD

**Issue by mjack3-tkh** [pytorch/vision#5034]
### 🚀 The feature

Now with GeneralizedRCNN.py in faster_rcnn.py and retinanet.py you can to specify

MIN_SIZE_TRAIN
MIN_SIZE_TRAIN_SAMPLING = [choice, range]
MAX_SIZE_TRAIN
MIN_SIZE_TEST
MAX_SIZE_TEST

where

- MIN_SIZE_TRAIN: Size of the smallest side of the image during training
- MIN_SIZE_TRAIN_SAMPLING: Sample size of smallest side by choice or random selection from range give by INPUT.MIN_SIZE_TRAIN
- MIN_SIZE_TEST: Size of the smallest side of the image during testing
- MAX_SIZE_TEST: Maximum size of the side of the image during testing

This is a similar as detectron2

### Motivation, pitch

I have been working in a project where needed to train in a complete range of resolutions and not testing in the biggest one of min attribute which is the default behaviur of the **GeneralizedRCNNTrasnform.py**. So, inplementing this new features allows you:

1. To use different inputs to train and testing with {MIN,MAX}_SIZE_TRAIN and {MIN,MAX}_SIZE_TEST attributes as Detectron2 does
2. Not just specify several input resolutions to train but ranges of resolution. In order to do this, the user can use MIN_SIZE_TRAIN and MIN_SIZE_TRAIN_SAMPLING = 'range' as Detectron2 does.

**Comment by datumbox** [comment]
@mjack3-tkh Thanks for the proposal.

I think with the introduction of the [prototype transforms](https://github.com/pytorch/vision/tree/main/torchvision/prototype/transforms), it might be interesting to investigate if moving the `GeneralizedRCNNTransform` transforms out of the model makes sense. Currently detection is the only family of models that does the resizing internally. Though this has benefits, it also makes the API different comparing to all Classification, Segmentation etc. At the moment it's not clear which direction we want to take, so I would be hesitant adding more functionality to the specific class.

For now, users are able to overwrite the specific functionality by overwriting the `model.transform` attribute after initializing the model.

**Comment by xiaohu2015** [comment]
@datumbox I think it is a very good proposal. The resize and normalization are included in `GeneralizedRCNNTransform`, which is not friendly to users.

**Comment by mjack3-tkh** [comment]
Thanks for you answer @datumbox

**Comment by datumbox** [comment]
Thanks @xiaohu2015 for the input. 

Agreed. The biggest concern for deprecating the `GeneralizedRCNNTransform`, is that it will be a massive breaking change. The detection models currently receive variable sized images. By removing the above transform and moving it out of the model, we effectively do something like what you describe; all transforms are applied prior to the model. This aligns the behaviour with every other model but it's a big change from the existing API. 

So we will need to think about the consequences to the users, get input from the community, create a plan on what to implement and how to roll it out etc. The new prototype Transforms being currently developed will help us a lot in that direction but since this is not finished/released yet, I would push any further discussion on refactoring detection in H1 of 2022 (possibly end Q1). Obviously I would love to get your input and I'll ping you when we send an RFC to debate this :)

**Comment by xiaohu2015** [comment]
@datumbox Thanks. If possible, I would like to take part in the refactoring work.

**Comment by mjack3-tkh** [comment]
same for me
